workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/EnvironmentScreenViewFactory.kt

```diff
@@ -1,18 +1,26 @@
 package com.squareup.workflow1.ui.container
 
-import com.squareup.workflow1.ui.DecorativeScreenViewFactory
+import com.squareup.workflow1.ui.Screen
 import com.squareup.workflow1.ui.ScreenViewFactory
+import com.squareup.workflow1.ui.ScreenViewFactory.Companion.forBuiltView
 import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
 import com.squareup.workflow1.ui.merge
+import com.squareup.workflow1.ui.toViewFactory
+import com.squareup.workflow1.ui.unwrapping
 
 @WorkflowUiExperimentalApi
-internal object EnvironmentScreenViewFactory : ScreenViewFactory<EnvironmentScreen<*>>
-by DecorativeScreenViewFactory(
-  type = EnvironmentScreen::class,
-  unwrap = { withEnvironment, inheritedEnvironment ->
-    Pair(
-      withEnvironment.screen,
-      inheritedEnvironment merge withEnvironment.viewEnvironment
-    )
+internal fun <WrappedT : Screen> EnvironmentScreenViewFactory():
+  ScreenViewFactory<EnvironmentScreen<WrappedT>> {
+  return forBuiltView { initialEnvScreen, initialEnvironment, context, container ->
```

Comment:
Not crazy about this name. `forBuilder`? `forConstructor`? `fromCode`?

Reply:
I liked `forXYZ` so much more than `bind` but now I'm jumping back on here to say I do love @RBusarow 's `from__` suggestion! or even `buildFrom__`

`buildFromCode` `buildFromLayoutResource` `buildFromBinding`